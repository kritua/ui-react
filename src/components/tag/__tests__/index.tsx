import React from 'react';
import { create } from 'react-test-renderer';

import { Tag } from '../';

it('Default', () => {
    const tree = create(
        <Tag children="Название тега" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Interactive', () => {
    const tree = create(
        <Tag
            children="Название тега"
            onClick={jest.fn()}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('presetSize', () => {
    const tree = create(
        <Tag
            children="Название тега"
            presetSize="small"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();

    tree.update(
        <Tag
            children="Название тега"
            presetSize="medium"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();

    tree.update(
        <Tag
            children="Название тега"
            presetSize="large"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
