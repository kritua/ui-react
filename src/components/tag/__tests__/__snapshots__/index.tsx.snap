// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default 1`] = `
<span
  className="tag tag_style-tag tag_size-small"
>
  Название тега
</span>
`;

exports[`Interactive 1`] = `
<span
  className="tag tag_style-tag tag_size-small tag_interactive"
  onClick={[MockFunction]}
>
  Название тега
</span>
`;

exports[`presetSize 1`] = `
<span
  className="tag tag_style-tag tag_size-small"
>
  Название тега
</span>
`;

exports[`presetSize 2`] = `
<span
  className="tag tag_style-tag tag_size-medium"
>
  Название тега
</span>
`;

exports[`presetSize 3`] = `
<span
  className="tag tag_style-tag tag_size-large"
>
  Название тега
</span>
`;
